### ISSUE TITLE
NET-2999 Fix S6672 FN: Mismatch in primary constructor is not recognized

### ISSUE THREAD

**Issue by jhinder** [SonarSource/sonar-dotnet#9784]
### Description

Currently only constructor declarations are checked. A logger type mismatch in a primary constructor is not yet recognized.

### Reproducer

```csharp
class A(ILogger<C> logger);
//             ^^^ no diagnostic here despite the mismatch

class B
{
    public B(ILogger<C> logger)
//                  ^^^ S6672: Update this logger to use its enclosing type.
    {
    }
}

class C;
```

### Product and Version

SonarQube for Visual Studio 9.4.0.16052

**Comment by pavel-mikula-sonarsource** [comment]
Hi @jhinder,

Thank you for reporting this case. I confirm it as False Negative
